tests/core/contracts/test_contract_constructor.py

```diff
@@ -142,32 +142,31 @@ def test_contract_constructor_build_transaction_to_field_error(MathContract):
         MathContract.constructor().buildTransaction({'to': '123'})
 
 
-def test_contract_constructor_build_transaction_nonce_field_error(MathContract):
-    with pytest.raises(ValueError):
-        MathContract.constructor().buildTransaction({'nonce': '123'})
-
-
 def test_contract_constructor_build_transaction_no_constructor(web3, MathContract):
+    nonce = web3.eth.getTransactionCount(web3.eth.coinbase) + 1
```

Comment:
Let's move this down to after sending the transaction so we don't have the confusing `+ 1`